router/dealer.go

```diff
@@ -951,6 +1000,53 @@ func (d *dealer) syncYield(callee *wamp.Session, msg *wamp.Yield, progress, canR
 		return false
 	}
 
+	// Check and handle Payload PassThru Mode
+	// @see https://wamp-proto.org/wamp_latest_ietf.html#name-payload-passthru-mode
+	if pptScheme, _ := msg.Options[wamp.OptPPTScheme].(string); pptScheme != "" {
+
+		// Let's check: was ppt feature announced by callee?
+		if !callee.HasFeature(wamp.RoleCallee, wamp.FeaturePayloadPassthruMode) {
+			// It's protocol violation, so we need to abort connection
+			//d.trySend(callee, &wamp.Error{
+			//	Type:    msg.MessageType(),
+			//	Request: msg.Request,
+			//	Details: wamp.Dict{},
+			//	Error:   wamp.ErrProtocolViolation,
+			//})
+			abortMsg := wamp.Abort{Reason: wamp.ErrProtocolViolation}
+			abortMsg.Details = wamp.Dict{}
+			abortMsg.Details[wamp.OptMessage] = "Peer is trying to use Payload PassThru Mode while it was not announced during HELLO handshake"
+			callee.Peer.Send(&abortMsg)
+			callee.Peer.Close()
+
+			return false
+		}
+
+		// Let's check if caller supports this feature
+		if !caller.HasFeature(wamp.RoleCaller, wamp.FeaturePayloadPassthruMode) {
+			d.trySend(callee, &wamp.Error{
+				Type:    msg.MessageType(),
+				Request: msg.Request,
+				Details: wamp.Dict{},
```

Comment:
no error details, sth like `caller does not support passthru mode` or do we not want to discolse information about the other end?

Reply:
yep, why not. Added.